### PR TITLE
chore: Initialize project

### PR DESCRIPTION
**Pull request by jkayani**
- Makefile to do the CI chores

- Cobra basic scaffolding for our CLI: https://github.com/spf13/cobra. I did it manually rather than via the code generator just to learn

- GH actions pipeline

Should we add something to help standup ArgoCD locally easily? We'll need that at some point to do testing of this plugin

### CODE REVIEW

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Basic pipeline
+on: push
```
Comment (werne2j) [inline .github/workflows/pipeline.yml:2]: Don't forget to add `pull requests` for when to run the pipeline

**Comment by jkayani** [comment]
Do we want to get advanced and configure merging to master to create GH releases?

File: .github/workflows/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
```
Comment (werne2j) [inline .github/workflows/pipeline.yml:9]: We will also want to run `go vet` as well as `go test`

Comment (jkayani) [reply .github/workflows/pipeline.yml:9]: Left out `go test` for now since we don't have any tests.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+VERSION=0.1
+OS_ARCH=darwin_amd64
+
+default: build
+
+format: 
+	go fmt github.com/IBM/argocd-vault-plugin/...
+
+quality:
+	go vet
```
Comment (werne2j) [inline Makefile:11]: I think you need to do `go vet ./...`

Comment (jkayani) [reply Makefile:11]: Yeah I had a suspicion but it didn't seem to do anything either way, thanks for telling me

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,24 @@
+BINARY=argocd-vault-plugin
+VERSION=0.1
+OS_ARCH=darwin_amd64
+
+default: build
+
+format: 
+	go fmt github.com/IBM/argocd-vault-plugin/...
```
Comment (werne2j) [inline Makefile:8]: I don't think we need this, unless we are stopping the build on code that is not in the correct format https://github.com/spf13/cobra/blob/master/Makefile#L16

Comment (jkayani) [reply Makefile:8]: You're right, I put that in there thinking it was a linter not a formatter

Comment (jkayani) [reply Makefile:8]: Do we want to fail on bad format?

Comment (werne2j) [reply Makefile:8]: I dont think we need that now. Most IDEs have auto formatting anyways.

Comment (jkayani) [reply Makefile:8]: Note for Josh: setup auto format in VS Code

**Review by werne2j — APPROVED** [review]

**Pull request merged**